command/bridge/finalize/finalize.go

```diff
@@ -274,3 +283,78 @@ func decodeGenesisAccounts(genesisSetRaw string) (map[types.Address]*contractsap
 
 	return genesisAccounts, nil
 }
+
+// validatorSetToABISlice converts given validators to generic map
+// which is used for ABI encoding validator set being sent to the rootchain contract
+func validatorSetToABISlice(o command.OutputFormatter,
+	validators []*validator.GenesisValidator) ([]*contractsapi.Validator, error) {
+	accSet := make(validator.AccountSet, len(validators))
+
+	if _, err := o.Write([]byte("[VALIDATORS - GATEWAY] \n")); err != nil {
+		return nil, err
+	}
+
+	for i, val := range validators {
+		if _, err := o.Write([]byte(fmt.Sprintf("%v\n", val))); err != nil {
+			return nil, err
+		}
+
+		blsKey, err := val.UnmarshalBLSPublicKey()
+		if err != nil {
+			return nil, err
+		}
+
+		accSet[i] = &validator.ValidatorMetadata{
+			Address:     val.Address,
+			BlsKey:      blsKey,
+			VotingPower: new(big.Int).Set(val.Stake),
+		}
+	}
+
+	hash, err := accSet.Hash()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := o.Write([]byte(
+		fmt.Sprintf("[VALIDATORS - GATEWAY] Validators hash: %s\n", hash))); err != nil {
+		return nil, err
+	}
+
+	return accSet.ToABIBinding(), nil
+}
+
+// initializeGateway initializes Gateway contract on external chain
+// based on finalized stake (voting power) of genesis validators on external
+func initializeGateway(outputter command.OutputFormatter,
```

Comment:
Just pass the `*config.Bridge` to the function so that we don't access the map every time we need some address.